### DISCUSSION TITLE
Paint on "Make overhangs printable"

### DISCUSSION THREAD

**Discussion by baile508** [OrcaSlicer/OrcaSlicer#3359] [Ideas]
Really like the added feature of "Make overhangs printable" but sometimes i dont want all the overhangs altered.  It would be awesome if you could paint the overhangs you want to be "printable" just like you paint the overhangs you want supported when you do manual supports.  I would even have the selections be the same as supports - Auto & Manual.  For Auto also be able to set a threshold angle just like in supports.

**Comment by DerrillWMS** [comment]
Or, I might suggest in the alternative, making the "Make overhangs printable" function ignore any areas where the "Support Painting" function has been applied (i.e., areas that are painted with support enforcer, or with support blocker). Currently, the "Support>Enable Support" checkbox applies 'across the board,' and so does the "Quality>Make overhangs printable" checkbox. However, the "Support Painting" function, on the other hand, applies only to areas specifically chosen by the user. And, given the purpose of the "Make overhangs printable" function is to eliminate support by changing the geometry of the object, it and the "Support Painting" function are mutually exclusive. So, by simply adding code in the "Make overhangs printable" function whereby it ignores any areas where the user has specifically painted where they want support enforced (no need to change the object's geometry there), or those areas where the user doesn't want support addressed at all (i.e., areas that really don't need support to print successfully, and changing the geometry there would be nonsensical), achieves the desired feature request.

**Comment by Elemecca** [comment]
Someone pointed out [in a comment on the original "make overhangs printable" feature request](https://github.com/SoftFever/OrcaSlicer/issues/1534#issuecomment-1837537204) that you can toggle the setting in a modifier object, so you can have "make overhangs printable" on globally and disable it with a modifier or vice versa. That's not as flexible as the support painter, but it works well enough for my model.
